group-coordinator/src/main/java/org/apache/kafka/coordinator/group/classic/ClassicGroup.java

```diff
@@ -1244,6 +1267,24 @@ public boolean completeSyncFuture(
         return false;
     }
 
+    /**
+     * Complete all the awaiting sync future with the give error.
+     *
+     * @param error  the error to complete the future with.
+     */
+    public void completeAllSyncFutures(
+        Errors error
+    ) {
+        members.forEach((__, member) -> completeSyncFuture(
+            member,
+            new SyncGroupResponseData()
+                .setProtocolName(protocolName.orElse(null))
+                .setProtocolType(protocolType.orElse(null))
+                .setAssignment(member.assignment())
```

Comment:
Do we have to set the assignment when returning an error? Same question for protocol name/type.

Reply:
We did set them in `resetAndPropagateAssignmentWithError(group, Errors.REBALANCE_IN_PROGRESS);` but I don't think they are necessary